src/test/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryServiceIT.java

```diff
@@ -231,4 +240,70 @@ public void stressTestCreateUrgentEvent()
     }
     sw.stop();
   }
+
+  private void generateDummyException(int vendorCode, SFSession session) throws SnowflakeSQLLoggedException
+  {
+    String queryID = "01234567-1234-1234-1234-00001abcdefg";
+    String reason = "This is a test exception.";
+    String sqlState = SqlState.NO_DATA;
+    throw new SnowflakeSQLLoggedException(queryID, reason, sqlState, vendorCode, session);
+  }
+  
+  /**
+   * Test case for checking telemetry message for SnowflakeSQLExceptions. Assert that telemetry OOB endpoint is reached
+   * after a SnowflakeSQLLoggedException is thrown.
+   * @throws SQLException
+   */
+  @Test
+  public void testSnowflakeSQLLoggedExceptionOOBTelemetry() throws SQLException, InterruptedException
+  {
+    // make a connection to initialize telemetry instance
+    Connection con = getConnection();
+    int fakeVendorCode = 27;
+    try
+    {
+      generateDummyException(fakeVendorCode, null);
+      fail();
+    }
+    catch (SnowflakeSQLLoggedException e)
+    {
+      // The error response has the same code as the the fakeErrorCode
+      assertThat("Communication error", e.getErrorCode(),
+              equalTo(fakeVendorCode));
+
+      // since it returns normal response,
+      // the telemetry does not create new event
+      Thread.sleep(WAIT_FOR_TELEMETRY_REPORT_IN_MILLISECS);
+      if (TelemetryService.getInstance().isDeploymentEnabled())
```

Comment:
It should be enabled by default right? We don't have to check this here.

Reply:
I think on some test systems, it is not enabled.